math/src/main/java/com/powsybl/math/timeseries/TimeSeriesTable.java

```diff
@@ -0,0 +1,597 @@
+                        double value2 = doubleBuffer.get(timeSeriesOffset2 + point);
+                        r[timeSeriesNum2] += (value1 - mean1) / stdDev1 * (value2 - mean2) / stdDev2;
+                    }
+                    r[timeSeriesNum2] /= tableIndex.getPointCount() - 1;
+                }
+            }
+        }
+    }
+
+    public double[] computePpmcc(String timeSeriesName, int version) {
+        int timeSeriesNum1 = doubleTimeSeriesNames.getIndex(timeSeriesName);
+        checkVersion(version);
+
+        Stopwatch stopWatch = Stopwatch.createStarted();
+
+        updateStatistics(version);
+
+        int statisticsIndex1 = getStatisticsIndex(version, timeSeriesNum1);
+        double stdDev1 = stdDevs[statisticsIndex1];
+
+        double[] r = new double[doubleTimeSeriesNames.size()];
+
+        if (stdDev1 == 0) { // constant time series
+            computeConstantTimeSeriesPpmcc(r, version);
+        } else {
+            computeVariableTimeSeriesPpmcc(r, timeSeriesNum1, statisticsIndex1, stdDev1, version);
+        }
+
+        LOGGER.info("PPMCC computed in {} ms", stopWatch.elapsed(TimeUnit.MILLISECONDS));
+
+        return r;
+    }
+
+    private static BufferedWriter createWriter(Path file) throws IOException {
+        if (file.getFileName().toString().endsWith(".gz")) {
+            return new BufferedWriter(new OutputStreamWriter(new GZIPOutputStream(Files.newOutputStream(file)), StandardCharsets.UTF_8));
+        } else {
+            return Files.newBufferedWriter(file, StandardCharsets.UTF_8);
+        }
+    }
+
+    public void writeCsv(Path file) {
+        try (BufferedWriter writer = createWriter(file)) {
+            writeCsv(writer, TimeSeriesConstants.DEFAULT_SEPARATOR, ZoneId.systemDefault());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public String toCsvString(char separator, ZoneId zoneId) {
+        try (StringWriter writer = new StringWriter()) {
+            writeCsv(writer, separator, zoneId);
+            writer.flush();
+            return writer.toString();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void writeHeader(Writer writer, CsvConfig config) throws IOException {
+        // write header
+        writer.write("Time");
+        writer.write(config.separator);
+        writer.write("Version");
+        for (TimeSeriesMetadata metadata : timeSeriesMetadata) {
+            writer.write(config.separator);
+            writer.write(metadata.getName());
+        }
+        writer.write(System.lineSeparator());
+    }
+
+    private class CsvCache {
+
+        static final int CACHE_SIZE = 10;
+
+        final double[] doubleCache = new double[CACHE_SIZE * doubleTimeSeriesNames.size()];
+
+        final String[] stringCache = new String[CACHE_SIZE * stringTimeSeriesNames.size()];
+    }
+
+    private static class  CsvConfig {
```

Comment:
Remove extra space between "class" and "CsvConfig"

Reply:
Done